diff --git a/packages/features/shell/Shell.tsx b/packages/features/shell/Shell.tsx
index 6c50c5c1b..8c8d1be32 100644
--- a/packages/features/shell/Shell.tsx
+++ b/packages/features/shell/Shell.tsx
@@ -13,6 +13,7 @@ import UnconfirmedBookingBadge from "@calcom/features/bookings/UnconfirmedBookin
 import ImpersonatingBanner from "@calcom/features/ee/impersonation/components/ImpersonatingBanner";
 import HelpMenuItem from "@calcom/features/ee/support/components/HelpMenuItem";
 import { TeamsUpgradeBanner } from "@calcom/features/ee/teams/components";
+import { useFlagMap } from "@calcom/features/flags/context/provider";
 import { KBarContent, KBarRoot, KBarTrigger } from "@calcom/features/kbar/Kbar";
 import TimezoneChangeDialog from "@calcom/features/settings/TimezoneChangeDialog";
 import { Tips } from "@calcom/features/tips";
@@ -22,6 +23,7 @@ import classNames from "@calcom/lib/classNames";
 import { APP_NAME, DESKTOP_APP_LINK, JOIN_SLACK, ROADMAP, WEBAPP_URL } from "@calcom/lib/constants";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import useTheme from "@calcom/lib/hooks/useTheme";
+import { isKeyInObject } from "@calcom/lib/isKeyInObject";
 import { trpc } from "@calcom/trpc/react";
 import useMeQuery from "@calcom/trpc/react/hooks/useMeQuery";
 import type { SVGComponent } from "@calcom/types/SVGComponent";
@@ -29,38 +31,39 @@ import {
   Button,
   Credits,
   Dropdown,
+  DropdownItem,
   DropdownMenuContent,
   DropdownMenuItem,
-  DropdownItem,
   DropdownMenuPortal,
   DropdownMenuSeparator,
   DropdownMenuTrigger,
   ErrorBoundary,
-  Logo,
   HeadSeo,
-  showToast,
+  Logo,
   SkeletonText,
+  showToast,
 } from "@calcom/ui";
 import {
-  FiMoreVertical,
-  FiMoon,
-  FiExternalLink,
-  FiLink,
-  FiSlack,
-  FiMap,
-  FiHelpCircle,
-  FiDownload,
-  FiLogOut,
+  FiArrowLeft,
+  FiArrowRight,
+  FiBarChart,
   FiCalendar,
   FiClock,
-  FiUsers,
+  FiDownload,
+  FiExternalLink,
+  FiFileText,
   FiGrid,
+  FiHelpCircle,
+  FiLink,
+  FiLogOut,
+  FiMap,
+  FiMoon,
   FiMoreHorizontal,
-  FiFileText,
-  FiZap,
+  FiMoreVertical,
   FiSettings,
-  FiArrowRight,
-  FiArrowLeft,
+  FiSlack,
+  FiUsers,
+  FiZap,
 } from "@calcom/ui/components/icon";
 
 import FreshChatProvider from "../ee/support/lib/freshchat/FreshChatProvider";
@@ -532,6 +535,11 @@ const navigation: NavigationItemType[] = [
     href: "/workflows",
     icon: FiZap,
   },
+  {
+    name: "Insights",
+    href: "/insights",
+    icon: FiBarChart,
+  },
   {
     name: "settings",
     href: "/settings/my-account/profile",
@@ -578,6 +586,8 @@ function useShouldDisplayNavigationItem(item: NavigationItemType) {
       trpc: {},
     }
   );
+  const flags = useFlagMap();
+  if (isKeyInObject(item.name, flags)) return flags[item.name];
   return !requiredCredentialNavigationItems.includes(item.name) || routingForms?.isInstalled;
 }
 
