diff --git a/packages/prisma/zod-utils.ts b/packages/prisma/zod-utils.ts
index c19a9824c..dbd964b7d 100644
--- a/packages/prisma/zod-utils.ts
+++ b/packages/prisma/zod-utils.ts
@@ -1,3 +1,4 @@
+import { Frequency as RRuleFrequency } from "rrule";
 import { z } from "zod";
 
 import { LocationType } from "@calcom/core/location";
@@ -11,6 +12,16 @@ export const eventTypeLocations = z.array(
   })
 );
 
+// Matching RRule.Options: rrule/dist/esm/src/types.d.ts
+export const recurringEvent = z.object({
+  dtstart: z.date().optional(),
+  interval: z.number().optional(),
+  count: z.number().optional(),
+  freq: z.nativeEnum(RRuleFrequency).optional(),
+  until: z.date().optional(),
+  tzid: z.string().optional(),
+});
+
 export const eventTypeSlug = z.string().transform((val) => slugify(val.trim()));
 export const stringToDate = z.string().transform((a) => new Date(a));
 export const stringOrNumber = z.union([z.string().transform((v) => parseInt(v, 10)), z.number().int()]);
